module/x/gravity/keeper/ethereum_event_vote.go

```diff
@@ -151,6 +151,18 @@ func (k Keeper) GetEthereumEventVoteRecord(ctx sdk.Context, eventNonce uint64, c
 	}
 }
 
+// DeleteEthereumEventVoteRecord deletes a vote record
+func (k Keeper) DeleteEthereumEventVoteRecord(ctx sdk.Context, eventVoteRecord *types.EthereumEventVoteRecord) {
+	event, err := types.UnpackEvent(eventVoteRecord.Event)
+	if err != nil {
+		panic(fmt.Sprintf("couldn't cast to event: %s", err))
+	}
+	if err := event.Validate(); err != nil {
+		panic(fmt.Sprintf("invalid event: %s", err))
+	}
```

Comment:
just a nit about this one that may be good to resolve in a different PR:
1. could it be ensured that `types.EthereumEventVoteRecord` contains only valid events, or instead of passing around these raw proto types, have their "safe" wrappers that are only constructed via a validation?
2. if not, could event unpacking do the validation as well? it seems a bit error-prone that it's two steps where I assume one will want both and may forget the second step

Reply:
1. Yes I guess having a generic interface for `EthereumEvent` and using any make it more flexible and eliminate some redundancy but we are losing in term of safety. Nevertheless, this is not something I would like to changing at this stage, at least for v2.

2. We can move the validation within the UnpackEvent. I'll do it in a separate PR and check that there is no side effect